src/Components/Search/SearchBar.tsx

```diff
@@ -116,79 +114,58 @@ export class SearchBar extends Component<Props, State> {
 
     if (entityType === "FirstItem") return
 
-    this.setState({
-      entityID,
-      entityType,
-    })
+    setEntityID(entityID)
+    setEntityType(entityType)
   }
 
-  @track((_props, _state, [query, hasResults]) => ({
+  track((_props, _state, [query, hasResults]) => ({
     action_type: hasResults
       ? DeprecatedSchema.ActionType.SearchedAutosuggestWithResults
       : DeprecatedSchema.ActionType.SearchedAutosuggestWithoutResults,
     query,
   }))
-  trackSearch(_term, _hasResults) {
+
+  const trackSearch = (_term, _hasResults) => {
     /* no-op */
   }
 
   // Throttled method to perform refetch for new suggest query.
-  throttledFetch = ({ value: term }) => {
-    const { relay } = this.props
-    const performanceStart = performance && performance.now()
-
-    relay.refetch(
-      {
-        hasTerm: true,
-        term,
-      },
-      null,
-      error => {
-        if (error) {
-          logger.error(error)
-          return
-        } else if (performanceStart && getENV("VOLLEY_ENDPOINT")) {
-          this.reportPerformanceMeasurement(performanceStart)
+  const throttledFetch = useCallback(
+    ({ value: term }) => {
+      const performanceStart = performance && performance.now()
+
+      relay.refetch(
+        {
+          hasTerm: true,
+          term,
+        },
+        null,
+        error => {
+          if (error) {
+            logger.error(error)
+            return
+          } else if (performanceStart && getENV("VOLLEY_ENDPOINT")) {
+            reportPerformanceMeasurement(performanceStart)
+          }
+          const edges = get(viewer, v => v.searchConnection.edges, [])
+          trackSearch(term, edges.length > 0)
         }
-        const { viewer } = this.props
-        // @ts-expect-error PLEASE_FIX_ME_STRICT_NULL_CHECK_MIGRATION
-        const edges = get(viewer, v => v.searchConnection.edges, [])
-        // @ts-expect-error PLEASE_FIX_ME_STRICT_NULL_CHECK_MIGRATION
-        this.trackSearch(term, edges.length > 0)
-      }
-    )
-  }
+      )
+    },
+    [relay, viewer]
+  )
 
-  componentDidMount() {
-    this.throttledFetch = throttle(this.throttledFetch, 500, {
+  useEffect(() => {
+    throttle(throttledFetch, 500, {
```

Comment:
you are using `throttle` incorrectly here (see the examples in [the Lodash documentation](https://lodash.com/docs/4.17.15#throttle)). You can also find usage examples in Eigen (like [this](https://github.com/artsy/eigen/blob/07007f404286907636c793c514cade5847c4fdef/src/app/Components/ArtworkFilter/Filters/KeywordFilter.tsx#L59), [this](https://github.com/artsy/eigen/blob/07007f404286907636c793c514cade5847c4fdef/src/app/Scenes/Artwork/Components/ImageCarousel/FullScreen/DeepZoom/DeepZoomLevel.tsx#L244), [this](https://github.com/artsy/eigen/blob/07007f404286907636c793c514cade5847c4fdef/src/app/Scenes/Search/Search2.tsx#L79), [this](https://github.com/artsy/eigen/blob/07007f404286907636c793c514cade5847c4fdef/src/app/Scenes/Search/components/SearchInput.tsx#L30) and etc)